File: buildSrc/src/main/groovy/org/stepik/gradle/plugins/idea/RunIdeaTask.java
package org.stepik.gradle.plugins.idea;

import org.stepik.gradle.plugins.jetbrains.BaseRunTask;

public class RunIdeaTask extends BaseRunTask {
    @Override
    public String[] getLibs() {
        return new String[] {
                "idea_rt.jar",
                "idea.jar",
                "bootstrap.jar",
                "extensions.jar",
                "util.jar",
                "openapi.jar",
                "trove4j.jar",
                "jdom.jar",
                "log4j.jar"
        };
    }

    @Override
    public String getPlatformPrefix() {
        return "Idea";
    }

}

File: buildSrc/src/main/kotlin/org/stepik/gradle/plugins/idea/RunIdeaTask.kt
package org.stepik.gradle.plugins.idea

import org.stepik.gradle.plugins.jetbrains.tasks.BaseRunTask


open class RunIdeaTask : BaseRunTask() {

    override fun getLibs() = arrayOf(
            "idea_rt.jar",
            "idea.jar",
            "bootstrap.jar",
            "extensions.jar",
            "util.jar",
            "openapi.jar",
            "trove4j.jar",
            "jdom.jar",
            "log4j.jar"
    )

    override val platformPrefix: String?
        get() {
            if (extension?.productType == "IC") {
                return "Idea"
            }
            return null
        }
}
